File: volleyball.client/src/entities/hall.ts
import Player from "./player";
import GameAgenda from './gameAgenda';
import HallCode from './hallCode';
import {PlayerLevel} from './playerLevel';

export default interface Hall {
    id: number,
    name: string,
    address: string
    administrator: Player
    isPublic: boolean,
    gamesAgenda?: GameAgenda[],
    minPlayers: number,
    maxPlayers: number,
    playersLevel:PlayerLevel,
    memberSeatsLimited:boolean,
    code?:HallCode,
    createGamesAutomatically:boolean
}

File: Volleyball.api/Enitities/Hall.cs
using System.Collections.Generic;

namespace Volleyball.api.Enitities
{
    public class Hall : IEntity
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Address { get; set; }
        public int MinPlayers { get; set; }
        public int MaxPlayers { get; set; }
        public bool IsPublic { get; set; } = true;
        public bool MemberSeatsLimited { get; set; } = true;
        public virtual PrivateHallCode Code { get; set; }
        public virtual Player Administrator { get; set; }
        public virtual IList<HallAgenda> Agendas { get; set; } = new List<HallAgenda>();
        public PlayerLevel PlayersLevel { get; set; }
        public bool CreateGamesAutomatically { get; set; }
    }
}
